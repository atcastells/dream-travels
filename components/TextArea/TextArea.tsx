import classNames from "classnames";
import { DetailedHTMLProps, InputHTMLAttributes } from "react";

export type TextAreaProps = DetailedHTMLProps<
  InputHTMLAttributes<HTMLTextAreaElement>,
  HTMLTextAreaElement
> & {
  label?: string;
  error?: string;
  containerProps?: DetailedHTMLProps<
    React.HTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  >;
};

export const TextArea = ({
  error,
  label,
  containerProps,
  ...inputProps
}: TextAreaProps) => {
  return (
    <div
      {...containerProps}
      className={classNames(
        "flex flex-col text-sm",
        containerProps?.className ? containerProps.className : ""
      )}
    >
      {label && <label className="text-sm mb-1">{label}</label>}
      <div
        className={classNames(
          "flex rounded-xl border bg-white border-gray-200 p-1",
          {
            "border-red-400": !!error,
          }
        )}
      >
        <textarea
          {...inputProps}
          className={classNames("bg-white text-black px-2 outline-none w-full")}
          rows={4}
        />
      </div>
      {error && <div className="text-red-400 text-xs mt-2 ml-4">{error}</div>}
    </div>
  );
};
