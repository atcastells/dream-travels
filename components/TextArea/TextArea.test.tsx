import { expect, test } from "@jest/globals";

import { render, screen } from "@testing-library/react";

import { TextArea } from "./TextArea";

test("renders text area", () => {
  const { baseElement } = render(<TextArea />);
  const input = screen.getByRole("textbox");
  expect(input).toBeTruthy();

  const label = screen.queryByRole("label");

  expect(label).toBeNull();

  const error = screen.queryByText("error");

  expect(error).toBeNull();

  //Snapshot testing

  expect(baseElement).toMatchSnapshot();
});
