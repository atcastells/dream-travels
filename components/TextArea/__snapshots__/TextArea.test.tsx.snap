// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text area 1`] = `
<body>
  <div>
    <div
      class="flex flex-col text-sm"
    >
      <div
        class="flex rounded-xl border bg-white border-gray-200 p-1"
      >
        <textarea
          class="bg-white text-black px-2 outline-none w-full"
          rows="4"
        />
      </div>
    </div>
  </div>
</body>
`;
